CIScripts/Test/PesterLogger/RemoteLogCollector.ps1

```diff
@@ -0,0 +1,71 @@
+function New-TrackedLogSource {
+    Param([Parameter(Mandatory = $true)] [string] $Path,
+          [Parameter(Mandatory = $false)] [PSSessionT[]] $Sessions)
+
+    $Sources = @()
+
+    $Sessions | ForEach-Object {
```

Comment:
We have thought of two alternatives for this loop.

One uses `foreach`: 

```powershell
If (-not $Sessions) {
  $Sessions = @($null)
}
foreach ($Session in $Sessions) {
  ...
}
```

Second one (it makes script analyzer suppression unnecessary):

```powershell
$Sources = $Sessions | ForEach-Object {
   ...

  $SingleSource
}
```

What do you think about them?